grand_challenge_forge/generation_utils.py

```diff
@@ -153,18 +151,27 @@ def copy_and_render(
                     _no_gpus=DEBUG,
                 )
 
-                # Write rendered content to output file (without .j2 extension)
-                output_file = output_file.with_suffix("")
-                with output_file.open("w") as f:
-                    f.write(rendered_content)
+                targetfile_zpath = output_file.with_suffix("")
 
-                # Copy permission bits
-                shutil.copymode(source_file, output_file)
-            else:
-                # Copy non-template files
-                shutil.copy2(source_file, output_file)
+                if targetfile_zpath.suffix == ".py":
+                    rendered_content = apply_black(rendered_content)
+
+                # Collect information about the file to be written to the zip file
```

Comment:
This and the time stuff below was added to fix the error you ran into right? Maybe add slightly more comments as to why this is necessary for future contributors?

Reply:
It partially fixes it. I think you are correct in that it is likely better to explicitly link to the issue at hand. I'll add a reference.